src/routes/Routine.tsx

```diff
@@ -6,31 +7,65 @@ import { formatDate } from '../lib/dateConversion';
 
 interface Iteration {
   start: string;
-}
+  running?: boolean;
+  end?: string;
+};
 
 interface RoutineData {
   name: string;
   iterations?: Iteration[];
-}
+};
 
 export async function loader({ params }: {params: Params}) {
   return jsonData.routines.find(routine => routine.name === params.routine);
-}
+};
 
 const Routine = () => {
   const routine = useLoaderData() as RoutineData;
+  const [iterations, setIterations] = useState<Iteration[]>(routine.iterations || []);
+
+  const handleOnStart = () => {
+    let runningRoutine = iterations.find(iteration => iteration.running) || false;
+
+    if (!runningRoutine) {
+      let currentDate = new Date();
+      let newRoutine = {"start": currentDate.toISOString(), "running": true};
+      setIterations([...iterations, newRoutine]);
+    }
+  };
+  
+  const handleOnEnd = () => {
+    let runningRoutine = iterations.find(iteration => iteration.running) || false;
+
+    if (runningRoutine) {
+      runningRoutine.running = false;
+      let currentDate = new Date();
+      runningRoutine.end = currentDate.toISOString();
```

Comment:
Why are you saving the end date as a string? What speaks against saving it as date time?